docs/secure-boot.md

```diff
@@ -1,39 +1,135 @@
-# `balenaOS` secure boot and disk encryption
+# `balenaOS` Secure Boot and Disk Encryption
 
-## UEFI (Unified Extensible Firmware Interface) support
+This document describes how Secure Boot and Disk Encryption are implemented in `balenaOS`. Technically these are two distinct functionalities:
+* Secure boot ensures only trusted operating system images can be booted on the device
+* Disk encryption ensures data is stored on the disk encrypted, rather than in plain form
 
-Systems that boot in UEFI mode and include a trusted platform module (TPM) can be configured to boot in secure boot mode.
+`balenaOS` ties these two together into a single feature in order to achieve operator-less unlocking of the encrypted data on trusted operating systems in a trusted state.
 
-Enabling secure boot also enables full disk encryption. Secure boot and disk encryption cannot be used separately as only booting in secure boot mode ensures the encrypted partitions cannot be mounted by untrusted software.
+## System Requirements
 
-Only TPM version 2 devices are currently supported.
+The only device type that currently supports Secure Boot is `generic-amd64`. In order to enable Secure Boot, the hardware needs to meet the following requirements:
```

Comment:
> The only device type that currently supports Secure Boot is `generic-amd64`

I would move this to the specific device repository and leave the generic conditions below only.

We could also think about adding it to the OS contract in some way as we will want to surface secure boot enabled devices.